docker/build-tools/Dockerfile

```diff
@@ -92,13 +92,22 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     xz-utils
 
 # Install protoc
-ADD https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-linux-x86_64.zip /tmp/
-RUN unzip /tmp/protoc-${PROTOC_VERSION}-linux-x86_64.zip
-RUN mv /tmp/bin/protoc ${OUTDIR}/usr/bin
+RUN set -eux; \
+    \
+    case $(uname -m) in \
```

Comment:
So it's build x64 on x64 host and arm64 on arm64 host?
Can you clarify in description that it's not crossing compiling?

Reply:
We could crossing compiling the `envoy` with wasm on same arch.
Like `wee8` the gn generate the compiling configration even using `uname -m`

so we have setup build env for both archs

For docker context, it build x64 on x64 context and arm64 on arm64 context.

Howerver, the backend hosts, depends how we create the buildx builder.

* one way with qemu, we could create a buildx builder in x64 host, and support build `linux/amd64` and  `linux/arm64` both, see more https://github.com/docker/buildx#building-multi-platform-images
  * but in this way, with very slow for compiling, executing is fine.
* another way could, create different nodes for different archs.

like
```bash
docker buildx create --name istio-builder --platform=linux/amd64 --node=istio-builder-amd64 --driver=xxx
docker buildx create --append --name istio-builder --platform=linux/arm64 --node=istio-builder-arm64 --driver=xxx
```

once we use `docker buildx builder --builder=istio-builder`, it will compile sub images for different archs on matched host.